src/NServiceBus.AcceptanceTests/Sagas/When_saga_is_mapped_to_complex_expression.cs

```diff
@@ -50,7 +50,10 @@ public class TestSaga02 : Saga<TestSagaData02>,
 
                 public Task Handle(OtherMessage message, IMessageHandlerContext context)
                 {
-                    Assert.AreEqual(Context.SagaId, Data.Id, "Existing instance should be found");
+                    if (Context.SagaId != Data.Id)
+                    {
+                        throw new AssertionException("Existing instance should be found");
```

Comment:
Why not move this out of the handler like the others? Won't this still just cause the handler to fail and not actually cause a failing assertion?